lua/entities/marums_hoverball/init.lua

```diff
@@ -4,36 +4,23 @@ AddCSLuaFile( "shared.lua" )  -- and shared scripts are sent.
 include('shared.lua')
  
 function ENT:Initialize()
-
-    --self:SetModel( "models/dav0r/hoverball.mdl" )
-    --self:SetModel( self.model )
-    self:PhysicsInit( SOLID_VPHYSICS )      -- Make us work with physics,
-    self:SetMoveType( MOVETYPE_VPHYSICS )   -- after all, gmod is a physics
-    self:SetSolid( SOLID_VPHYSICS )         -- Toolbox
+    self.wireCInputs = { "Hover Distance", "Hover Force", "Air Resistance", "Rotation Damping", "Detect Water"}
+    self:PhysicsInit( SOLID_VPHYSICS )      
+    self:SetMoveType( MOVETYPE_VPHYSICS )   
+    self:SetSolid( SOLID_VPHYSICS )     
     self:SetCollisionGroup(COLLISION_GROUP_DISSOLVING)
     self.delayedForce = 0
-    --self.hoverdistance = cvars.Number( "mhb_height" )
-    --self.hoverforce = cvars.Number( "mhb_force" )
-    --self.damping = cvars.Number( "mhb_air_resistance" )
-    --self.rotdamping = cvars.Number( "mhb_angular_damping" )
-    --self.detectswater = cvars.Bool( "mhb_detects_water" )
     self.mask = MASK_NPCWORLDSTATIC
-    --self.constrainedEntities = constraint.GetAllConstrainedEntities( self )
-    --if (istable(self.constrainedEntities)) then
-    --  table.insert(self.constrainedEntities, self)
-    --else
-    --  --self.constrainedEntities = {self}
-    --end
-    if self.detectswater then
-        self.mask = self.mask+MASK_WATER
-    end
-
+    self:GetOwner():AddCleanup( "wire_marums_hoverball", entity )
     local phys = self:GetPhysicsObject()
-    if phys:IsValid() then
+    if (phys:IsValid()) then
```

Comment:
why the redundant parentheses 😕

Reply:
You annoyed yet? I may have wanted to make sure it was done right.